planner/optimize.go

```diff
@@ -123,15 +131,11 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 			fp = plannercore.TryFastPlan(sctx, node)
 		}
 		if fp != nil {
-			if !useMaxTS(sctx, fp) {
-				if err := sessiontxn.WarmUpTxn(sctx); err != nil {
-					return nil, nil, err
-				}
-			}
 			return fp, fp.OutputNames(), nil
 		}
 	}
-	if err := sessiontxn.WarmUpTxn(sctx); err != nil {
+
+	if err = sessiontxn.WarmUpTxn(sctx); err != nil {
```

Comment:
Is it possible to send too many uncessary tso requests?

Reply:
No, when `WarmUpTxn` is called, the `baseTxnContextProvider.prepareTxn` will be called to start tso fetching. `prepareTxn` is safe for called multiple times, it will return without doing anything when tso already prepared

Reply:
I mean in case a tso request is not needed, e.g. when using max_uint64.